public/app/features/manage-dashboards/state/RecentlyDeletedActions.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+
+import { Button } from '@grafana/ui';
+
+import appEvents from '../../../core/app_events';
+import { Trans } from '../../../core/internationalization';
+import { useDispatch } from '../../../types';
+import { ShowModalReactEvent } from '../../../types/events';
+import { useRestoreDashboardMutation } from '../../browse-dashboards/api/browseDashboardsAPI';
+import { setAllSelection, useActionSelectionState } from '../../browse-dashboards/state';
+import { RestoreModal } from '../components/RestoreModal';
+import { useRecentlyDeletedStateManager } from '../utils/useRecentlyDeletedStateManager';
+
+export function RecentlyDeletedActions() {
+  const dispatch = useDispatch();
+  const selectedItems = useActionSelectionState();
+  const [_, stateManager] = useRecentlyDeletedStateManager();
+  const isSearching = stateManager.hasSearchFilters();
+
+  const [restoreDashboard, { isLoading: isRestoreLoading }] = useRestoreDashboardMutation();
+
+  const onActionComplete = () => {
+    dispatch(setAllSelection({ isSelected: false, folderUID: undefined }));
+
+    if (isSearching) {
+      stateManager.doSearchWithDebounce();
+    }
+  };
+
+  const onRestore = async () => {
+    const promises = Object.entries(selectedItems.dashboard)
+      .filter(([_, selected]) => selected)
+      .map(async ([uid]) => {
+        return await restoreDashboard({ dashboardUID: uid });
+      });
```

Comment:
```suggestion
      .map(([uid]) => restoreDashboard({ dashboardUID: uid }));
```
There's no need for async/await here, it only creates an extra promise wrapper around `restoreDashboard`.